### PR TITLE
Adding to the documentation

### PR DESCRIPTION
**Pull request by karland**
I red [#219](https://github.com/metalsmith/metalsmith/issues/219) and thought I'll give it a try...

It is certainly not complete, but maybe it is a step into the right direction. Hope it helps.

Please read carefully, to make sure that everything is correct. Also, I am not a native speaker, so you might find some errors....

### CODE REVIEW

**Comment by Ajedi32** [comment]
Yeah, this is definitely a step in the right direction. I don't have time to dive into this too deep right now, but I really like where this is headed.

A few general comments on the overall structure of the page:

Before diving too deep into the philosophy behind Metalsmith and how it works, I think it may be best to give a quick "elevator pitch" first about what makes Metalsmith unique. The idea is to give newcomers a quick taste of what Metalsmith is before getting them bogged down in the details of how to actually use it to build a site.

I'd move the install steps to below that elevator pitch too. The first thing a new user is going to want to know about Metalsmith is what it is and whether or not it's something they're interested in using. Only after they've determined that the want to use Metalsmith will they care how to install it. Ideally we'd also include a quick "Hello world!" example of how to get up and running with the simplest possible site after the install instructions, that's something we're really lacking right now.

The details about how Metalsmith works are great! One minor mistake I did notice is that the page is currently claiming that Metalsmith transforms the directory into an "array" of JavaScript objects. It's actually more of an "object" of JavaScript objects (not really sure on the best way to word that). E.g. If you have `file1.md` and `file2.md` Metalsmith transforms that into something like:

```javascript
{
  "file1.md": {
    title: 'A Catchy Title',
    contents: new Buffer('An unfinished article...')
  },
  "file2.md": {
    title: 'A Catchy Title',
    contents: new Buffer('An unfinished article...')
  }
}
```

Not like:

```javascript
[
  {
    "file1.md": {
      title: 'A Catchy Title',
      contents: new Buffer('An unfinished article...')
    }
  },
  {
    "file2.md": {
      title: 'A Catchy Title',
      contents: new Buffer('An unfinished article...')
    }
  }
]
```

There were also a few minor spelling and grammar issues I noticed, but rather than point them all out in this PR I think I'd prefer to focus on the larger issues here. It'll be easier for me to fix those types of issues directly (with a commit) anyway rather than through a bunch of comments.

**Comment by karland** [comment]
Hey @Ajedi32 thanks for the positive feedback. I am happy to work some more on the documentation, but it would be great if you could indicate how to principally move forward. Let me address your comments one after another:

+ You say, "I don't have time to dive into this too deep right now". I am not sure exactly what this means. Does it means that you will not have the time to review the documentation in the near future, so it just stays a PR?

+ My idea of improving the documentation is a step-by-step approach. People may have time for small improvements but not for huge steps. I love metalsmith. It is even better than I orginally thought, but the lack of documentation almost put me off. What convinced me was that according to [http://www.staticgen.com/](http://www.staticgen.com/) it was so successful, so I pulled through. Amongst the generators which are JS based, I am convinced Metalsmith is probably the most flexible one.

+ I have started the addition to the documentation from a more experienced point of view. I have used and am still using [Middleman](https://middlemanapp.com/) --- which is also great. However, I wanted to use my static site generator alongside Gulp and also reduce the complexity of my workflow to only using JS. With this objectiv, at first I was struggling with understanding the concept of Metalsmith, which is why I added the explanations. But I agree, this is probably more for people who want to migrate from a different static site generator and also want to exploit all current possiblities of node.js

+ Nevertheless, I agree with the Elevator pitch. I have therefore added some Welcome statement? For this I have also updated the [example static site generator](https://github.com/metalsmith/metalsmith/tree/master/examples/static-site) in this [PR 235](https://github.com/metalsmith/metalsmith/pull/235). 

+ Concerning the array vs JS object of JS objects, you are right. It's not an array. I have changed that already. I have also added your example code of how this JS object of JS objects looks like to the PR.

Well, finally it is your call. If you think that with this PR the website is better than before you should accept it even though you may think it is not quite there yet. I would be more than happy if you remove the language mistakes. I am also pretty easy if you add your changes. My believe is that the community will grow with better documentation and then also help to improve the documentation.

**Comment by Ajedi32** [comment]
> You say, "I don't have time to dive into this too deep right now". I am not sure exactly what this means.

It just means I need to set aside some time to look at it, make a few grammar corrections, and build the site on my PC to make sure everything looks good before I merge. I've got final exams happening at college these next couple weeks so it may be a while before I have a chance to do that, but I'll get to it.

> My idea of improving the documentation is a step-by-step approach. People may have time for small improvements but not for huge steps.

Yeah, sorry if I made it sound like I was expecting a complete overhaul. Reading this PR made me think about a lot of other specific ways the docs could be improved, and I wrote down a lot of those thoughts in my above comment - I wasn't necessarily expecting _all_ of those changes to be addressed by this PR.

**Comment by karland** [comment]
Sounds good. And good luck with your exams. So, I leave you to it for a review. I made one final change for the moment. I felt that the [`src/index.md`](https://github.com/metalsmith/metalsmith.io/blob/master/src/index.md) was difficult to maintain with regard to the `<code>` parts, so I changed that and introduced [`hghlight.js`](https://highlightjs.org/static/demo/) for code highlighting. For the stylesheet I chose [`tomorrow.min.css`](https://github.com/isagalaev/highlight.js/tree/master/src/styles) to keep the background white and to have a "modern" look.

**Comment by karland** [comment]
Hi @Ajedi32, hope your exams went well. Any suggestions for my PR and commits added? Cheers.

File: build/build.css [hunk 1]
```diff
@@ -753,6 +753,34 @@ hr:nth-of-type(6) {
   background-image: url("metalsmith.io/images/divider-6.png");
 }
 
+hr:nth-of-type(7) {
+  background-image: url("metalsmith.io/images/divider-1.png");
+}
+
```
Comment (Ajedi32) [inline build/build.css]: Interesting. I didn't even realize `build/` was checked in to git. @lambtron, how are builds normally handled for this site?

**Comment by Ajedi32** [comment]
Cool, looking pretty good. I hadn't noticed you'd been adding more commits to this. My exams went pretty well, I'll probably have some time to look this over and get it merged this weekend.

File: build/build.css (hunk 1 above)
Comment (karland) [inline build/build.css]: Yeah, I think the `build/` should be removed. I just added to the static-site example in metalsmith and suggested to gitignore build:
[metalsmith PR #237](https://github.com/metalsmith/metalsmith/pull/237/commits/51a8f80c6a3871f65e0c58ec8f47525e051c0ee2)

**Comment by karland** [comment]
Sounds good.

**Comment by karland** [comment]
@Ajedi32 As you see I have added also the official API from the [metalsmith Readme.md](https://github.com/metalsmith/metalsmith/blob/master/Readme.md) to the documentation. Mind you, [PR #229](https://github.com/metalsmith/metalsmith/pull/229) is not yet contained.

**Comment by Ajedi32** [comment]
Thanks. Sorry I haven't gotten to this yet; I promise I haven't forgotten about it. :wink:

**Comment by karland** [comment]
Hi @Ajedi32 what is your status on that? Cheers.

**Comment by Ajedi32** [comment]
Yeah, sorry this is taking so long. I do still intend to get to this (it's on my list), but Metalsmith has taken a back seat to other priorities in my life lately.

I'm certainly not the only member on this project so if someone else can find the time to address this PR that'd be great. Otherwise, like I said, it's still on my list and I do intend to get to it eventually.

**Comment by karland** [comment]
@Ajedi32 thanks for the update. I understand that you have other priorities, however, metalsmith is such great tool, that I hope that members can attend to PRs and keep the project up to date. Best regards.

**Comment by fourpixels** [comment]
Come on guys, @karland spent his time doing this and there's no one to even merge it? What kind of project is this - if you have abandoned it at least add someone to the organisation that will keep developing it! The PR is opened May 7th, which is like almost 4 months ago! I started to feel this one won't work out..

**Comment by woodyrew** [comment]
@karland I think we're in a position to get your changes added. Could you rebase to resolve the conflicts?

**Comment by karland** [comment]
Rebasing it in [PR #205](https://github.com/metalsmith/metalsmith.io/pull/205)

**Pull request closed without merging**